### ISSUE TITLE
oxlint -c ./eslintrc.json ignores "error" rules

### ISSUE THREAD

**Issue by CalinaCristian** [oxc-project/eslint-plugin-oxlint#16]
I added the `.eslintrc.json` file, add it sems that it works to turn off some rules but not to enhance them to error instead of warning

For example:

```
{
    "rules": {
        "react/jsx-key": "off",
        "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
        "unicorn/no-new-array": "off",
        "no-console": "error"      
    }
}
```

Everything is turned off as expected, but `no-console` is still reported as a warn.

**Comment by RodrigoTomeES** [comment]
any updates about this?

**Comment by Djiit** [comment]
I'd love to have a reproduction example repo to work on it.

**Comment by folkvir** [comment]
Very easy to reproduce:

.eslintrc.json
```
{
  "rules": {
    "no-null": "error"
  }
}
```
myfile.ts
```ts
let foo = null;
```

`oxlint --config .eslintrc.json myfile.ts`  reports  `eslint-plugin-unicorn(no-null): Disallow the use of the 'null' literal` as a warning while it should be reported as an error.

**Comment by Sysix** [comment]
I wanted to look into this bug. I used your example code snippet and im getting now a error:

```bash
npx oxlint --config .eslintrc.json test.ts
npm verbose cli /home/sysix/.nvm/versions/node/v22.1.0/bin/node /home/sysix/.nvm/versions/node/v22.1.0/lib/node_modules/npm/bin/npm-cli.js
npm info using npm@10.7.0
npm info using node@v22.1.0
npm verbose title npm exec oxlint --config .eslintrc.json test.ts
npm verbose argv "exec" "--" "oxlint" "--config" ".eslintrc.json" "test.ts"
npm verbose logfile logs-max:10 dir:/home/sysix/.npm/_logs/2024-10-08T18_02_58_066Z-
npm verbose logfile /home/sysix/.npm/_logs/2024-10-08T18_02_58_066Z-debug-0.log
npm http fetch GET 200 https://registry.npmjs.org/oxlint 191ms (cache revalidated)

  × eslint(no-console): Unexpected console statement.
   ╭─[test.ts:1:9]
 1 │ console.log('test#');
   ·         ───
   ╰────

Finished in 16ms on 1 file with 97 rules using 24 threads.
Found 0 warnings and 1 error.
npm verbose exit 1
npm verbose code 1
```

@CalinaCristian can u reproduce it with the newest version?
